File: graal-nodejs/test/graal/unit/integer.js
/*
 * Copyright (c) 2015, Oracle and/or its affiliates. All rights reserved.
 * ORACLE PROPRIETARY/CONFIDENTIAL. Use is subject to license terms.
 */

var assert = require('assert');
var module = require('./_unit');

describe('Integer', function () {
    describe('New', function () {
        it('should be within int32_t type', function () {
            var two31 = Math.pow(2, 31);
            assert.strictEqual(module.Integer_New(-1), -1);
            assert.strictEqual(module.Integer_New(0), 0);
            assert.strictEqual(module.Integer_New(1), 1);
            assert.strictEqual(module.Integer_New(two31 - 1), two31 - 1);
            assert.strictEqual(module.Integer_New(two31), -two31);
        });
    });
    describe('NewFromUnsigned', function () {
        it('should be within uint32_t type', function () {
            var two32 = Math.pow(2, 32);
            assert.strictEqual(module.Integer_NewFromUnsigned(-1), two32 - 1);
            assert.strictEqual(module.Integer_NewFromUnsigned(0), 0);
            assert.strictEqual(module.Integer_NewFromUnsigned(1), 1);
            assert.strictEqual(module.Integer_NewFromUnsigned(two32 - 1), two32 - 1);
            assert.strictEqual(module.Integer_NewFromUnsigned(two32), 0);
        });
    });
});

File: graal-nodejs/test/graal/unit/integer.cc
/*
 * Copyright (c) 2015, Oracle and/or its affiliates. All rights reserved.
 * ORACLE PROPRIETARY/CONFIDENTIAL. Use is subject to license terms.
 */

#define SUITE Integer

// Integer::New

EXPORT_TO_JS(New) {
    Isolate* isolate = args.GetIsolate();
    int32_t value = (int32_t) args[0]->ToInt32()->Value();
    args.GetReturnValue().Set(Integer::New(isolate, value));
}

// Integer::NewFromUnsigned

EXPORT_TO_JS(NewFromUnsigned) {
    Isolate* isolate = args.GetIsolate();
    uint32_t value = (uint32_t) args[0]->ToUint32()->Value();
    args.GetReturnValue().Set(Integer::NewFromUnsigned(isolate, value));
}

#undef SUITE
